### PR TITLE
New port types

### PR DESCRIPTION
**Pull request by rossella**

### CODE REVIEW

File: src/midonetclient/api.py [hunk 1]
```diff
@@ -222,10 +235,116 @@ def delete_trace_messages(self, id_):
...
+            "inv_dl_src": False,
+            "dl_dst": None,
+            "inv_dl_dst": False,
+            "nw_proto": None,
+            "inv_nw_proto": False,
+            "dl_type": None,
+            "inv_dl_type": False,
+            "jump_chain_id": None,
+            "jump_chain_name": None,
+          "match_forward_flow": False,
```
Comment (admytren) [inline src/midonetclient/api.py]: Indentation

File: src/midonetclient/api.py~ [hunk 2]
```diff
@@ -0,0 +1,415 @@
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
```
Comment (admytren) [inline src/midonetclient/api.py~]: This file should not be in the repository - it's a backup of api.py

File: src/midonetclient/application.py~ [hunk 3]
```diff
@@ -0,0 +1,229 @@
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
```
Comment (admytren) [inline src/midonetclient/application.py~]: This file should not be checked in

File: src/midonetclient/dhcp_subnet.py~ [hunk 4]
```diff
@@ -0,0 +1,62 @@
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
```
Comment (admytren) [inline src/midonetclient/dhcp_subnet.py~]: This file should not be in the archive

File: src/midonetclient/midoapi_exceptions.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2013 Midokura PTE LTD.
```
Comment (admytren) [inline src/midonetclient/midoapi_exceptions.py]: Is there any reason of wrapping this exception? Also recommend moving this code to "exc.py"

File: src/midonetclient/port_type.py~ [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
```
Comment (admytren) [inline src/midonetclient/port_type.py~]: Do not check in this file

File: src/midonetclient/resource_base.py [hunk 7]
```diff
@@ -66,8 +66,8 @@ def get_children(self, uri, query, headers, clazz, extra_args=None):
                                          headers=headers)
```
Comment (admytren) [inline src/midonetclient/resource_base.py:66]: Looks like an unnecessary change - space deleted on one line added on the other?

File: src/midonetclient/rule.py~ [hunk 8]
```diff
@@ -0,0 +1,293 @@
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
```
Comment (admytren) [inline src/midonetclient/rule.py~]: Do not check in this file

File: src/midonetclient/trace_condition.py [hunk 9]
```diff
@@ -22,6 +22,7 @@
 from midonetclient import vendor_media_type
 from midonetclient.resource_base import ResourceBase
 
+
```
Comment (admytren) [inline src/midonetclient/trace_condition.py:25]: Whitespace

File: src/midonetclient/vendor_media_type.py~ [hunk 10]
```diff
@@ -0,0 +1,105 @@
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
```
Comment (admytren) [inline src/midonetclient/vendor_media_type.py~]: Do not check in this file

File: src/midonetclient/vlan_aware_bridge.py~ [hunk 11]
```diff
@@ -0,0 +1,121 @@
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
```
Comment (admytren) [inline src/midonetclient/vlan_aware_bridge.py~]: Do no check in this file

File: src/midonetclient/vlan_bridge.py~ [hunk 12]
```diff
@@ -0,0 +1,87 @@
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
```
Comment (admytren) [inline src/midonetclient/vlan_bridge.py~]: Do not check in this file

File: src/midonetclient/vlan_bridge_interior_port.py~ [hunk 13]
```diff
@@ -0,0 +1,66 @@
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
```
Comment (admytren) [inline src/midonetclient/vlan_bridge_interior_port.py~]: Do not check in this file

File: src/midonetclient/vlan_bridge_port.py~ [hunk 14]
```diff
@@ -0,0 +1,57 @@
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
```
Comment (admytren) [inline src/midonetclient/vlan_bridge_port.py~]: Do not check in this file

File: src/midonetclient/vlan_bridge_trunk_port.py~ [hunk 15]
```diff
@@ -15,8 +15,7 @@
 # License for the specific language governing permissions and limitations
```
Comment (admytren) [inline src/midonetclient/vlan_bridge_trunk_port.py~]: Do not check in this file - this should have been a remove operation, not move

File: src/midonetclient/api.py [hunk 16]
```diff
@@ -20,14 +20,23 @@
 # @author: Artem Dmytrenko <art@midokura.com>, Midokura
 
 import logging
+from socket import error as socket_error
 
 from midonetclient import auth_lib
 from midonetclient.application import Application
+from midonetclient import midoapi_exceptions
```
Comment (admytren) [inline src/midonetclient/api.py:28]: whitespace

File: src/midonetclient/api.py [hunk 17]
```diff
@@ -20,14 +20,23 @@
...
 
 
 LOG = logging.getLogger(__name__)
```
Comment (admytren) [inline src/midonetclient/api.py]: whitespace

File: src/midonetclient/trace_condition.py (hunk 9 above)
Comment (rossella) [inline src/midonetclient/trace_condition.py:25]: where is it? can't see it. 2 lines before a class are ok

File: src/bin/midonet-cli [hunk 18]
```diff
@@ -1576,12 +1576,19 @@ class Condition(ObjectType):
...
                                                     'match_return_flow') ],
                               optional = True))
-        self.put_attr(SingleAttr(name = 'port-group',
-                                 value_type = ObjectRef('port-group'),
-                                 getter = 'get_port_group',
-                                 setter = 'port_group',
-                                 inv_getter = 'is_inv_port_group',
-                                 inv_setter = 'inv_port_group',
+        self.put_attr(SingleAttr(name = 'port-group-src',
+                                 value_type = ObjectRef('port-group-src'),
```
Comment (gontanon) [inline src/bin/midonet-cli]: IIRC the argument for ObjectRef() constructor should be a collection name at the top level of the object tree, that would make this line ObjectRef('port-group'), meaning that you are referencing an object in the top-level 'port-group' collection.

File: src/bin/midonet-cli [hunk 19]
```diff
@@ -1576,12 +1576,19 @@ class Condition(ObjectType):
...
+                                 getter = 'get_port_group_src',
+                                 setter = 'port_group_src',
+                                 inv_getter = 'is_inv_port_group_src',
+                                 inv_setter = 'inv_port_group_src',
+                                 optional = True))
+        self.put_attr(SingleAttr(name = 'port-group-dst',
+                                 value_type = ObjectRef('port-group-dst'),
```
Comment (gontanon) [inline src/bin/midonet-cli]: ditto.

File: src/midonetclient/trace_condition.py (hunk 9 above)
Comment (admytren) [inline src/midonetclient/trace_condition.py:25]: That's fine, you can ignore my comment :)

File: src/bin/midonet-cli (hunk 18 above)
Comment (rossella) [inline src/bin/midonet-cli]: Thanks!
